pkg/interactive/docker.go

```diff
@@ -76,39 +74,36 @@ func (d *Docker) AddMount(typeStr, source, target string, optAdditionalArgs ...s
 	d.AddArgs("--mount", fmt.Sprintf("type=%s,source=%s,target=%s%s", typeStr, source, target, addl))
 }
 
-// AddRWOverlay mounts a directory into the image at the desired location, but with an overlay
-//  so internal changes do not modify the external directory.
-// externalDirectory probably needs to be an absolute path
-// Returns a function to clean up the mount (but does not delete the directory).
-// Uses sudo and probably only works on Linux
-func (d *Docker) AddRWOverlay(externalDirectory, internalDirectory string) func() {
-	tmpDir, err := ioutil.TempDir("", "overlay")
+// CopyAndAddMount copies the source files into a temp directory, and then
+// mounts them as the target. It also returns a function to remove the temp
+// directory for cleaning up.
+func (d *Docker) CopyAndAddMount(typeStr, parentDir, source, target string, optAdditionalArgs ...string) (func(), error) {
+	fi, err := os.Stat(source)
 	if err != nil {
-		log.Fatal(err)
+		return nil, fmt.Errorf("error getting the FileInfo for %q: %w", source, err)
 	}
-	subDirs := []string{"upper", "work", "overlay"}
-	for _, d := range subDirs {
-		err = os.Mkdir(path.Join(tmpDir, d), 0777)
-		if err != nil {
-			log.Fatal(err)
-		}
+
+	tempDir, err := ioutil.TempDir(parentDir, fi.Name())
+	if err != nil {
+		return nil, fmt.Errorf("error creating a temporary directory: %w", err)
+	}
+	if err := os.Chmod(tempDir, 0777); err != nil {
```

Comment:
Why not `0770`, given it's a directory?

Reply:
I meant `0666`

Reply:
Changed to `0666`.

Reply:
um, directories need to be executable and we're writing to it, so why isn't it `0777`?

Reply:
Cory is correct. Executable flag for directories grant permission on its content. Discard this comment.

Reply:
Yeah I just tried `0666` and it will actually throw out an error. I've reverted the change.